execution_engine/src/core/engine_state/engine_config.rs

```diff
@@ -95,4 +153,69 @@ impl EngineConfig {
     pub fn strict_argument_checking(&self) -> bool {
         self.strict_argument_checking
     }
+
+    /// Get the engine config's administrative accouAnts.
+    #[must_use]
+    pub fn administrative_accounts(&self) -> Vec<AdministratorAccount> {
+        self.administrative_accounts.lock().unwrap().clone()
+    }
+
+    /// Checks if chain is configured in private mode.
+    #[must_use]
+    pub fn is_private_chain(&self) -> bool {
+        !self.administrative_accounts().is_empty()
+    }
+
+    /// Get the engine config's allow auction bids.
+    #[must_use]
+    pub fn allow_auction_bids(&self) -> bool {
+        self.allow_auction_bids
+    }
+
+    /// Get the engine config's allow unrestricted transfers.
+    #[must_use]
+    pub fn allow_unrestricted_transfers(&self) -> bool {
+        self.allow_unrestricted_transfers
+    }
+
+    /// Checks if an account hash is an administrator.
+    ///
+    /// This method returns a `None` if chain is not configured in a private chain. Otherwise Some
+    /// with a value whether passed account hash is an admin.
+    #[must_use]
+    pub(crate) fn is_account_administrator(&self, account_hash: &AccountHash) -> Option<bool> {
+        let borrowed_admins = self.administrative_accounts.lock().unwrap();
```

Comment:
Agreed this was unnecessary and risky (and ugly). Since we don't need to mutate, we don't need this monstrous Arc<Mutex<Vec<T>>>  to be able to mutate it at upgrade. When `commit_upgrade` executes, we already had a new chain spec, and therefore all of that was unnecessary.